Screenboard for the nomad integration

steveny91 opened the pull request:
### What does this PR do?

To attach an out of the box overview screenboard for the nomad integration

### Motivation

Assigned as a embedding onboarding task.

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

N/A

steveny91 commented:
Staging link:
https://dd.datad0g.com/screen/integration/30327/nomad-overview

Screenshot from sandbox:
https://a.cl.ly/z8u86E1R

ChristineTChen commented on nomad/assets/dashboards/overview.json:
```diff
@@ -0,0 +1,867 @@
+            "y": 84
+        },
+        {
+            "bgcolor": "white",
```
```suggestion
            "bgcolor": "gray",
```

Lets keep section headers as gray. It makes it easier to read the different categories of graphs

ChristineTChen commented on nomad/assets/dashboards/overview.json:
```diff
@@ -0,0 +1,867 @@
+            "y": 0
+        },
+        {
+            "bgcolor": "white",
```
```suggestion
            "bgcolor": "gray",
```

ChristineTChen commented on nomad/assets/dashboards/overview.json:
```diff
@@ -0,0 +1,867 @@
+            "y": 84
+        },
+        {
+            "bgcolor": "white",
```
```suggestion
            "bgcolor": "gray",
```

ChristineTChen commented on nomad/assets/dashboards/overview.json:
```diff
@@ -0,0 +1,867 @@
+            "y": 52
+        },
+        {
+            "bgcolor": "white",
```
`Consul metrics` can be either gray or pink.

Pink is usually the color we use to reference addtl integration metrics but since these are emitted by nomad, they can also just be a category.
https://datadoghq.dev/integrations-core/guidelines/dashboards/#other-technologies

ChristineTChen commented on nomad/assets/dashboards/overview.json:
```diff
@@ -0,0 +1,867 @@
+            "y": 52
+        },
+        {
+            "bgcolor": "white",
```
Same here with `Consul Metrics`. Can consider raft metrics as other technology.

Also quick question, if customers were to receive consul metrics, would raft section be empty? If so could we move the `Runtime Metrics` section more to the left since those are known to be emitted?

steveny91 commented:
Addressed the changes and moved the additional metrics around as you suggested. Here's a new screenshot from my sandbox. 

https://a.cl.ly/NQugw9lY

ChristineTChen commented on nomad/assets/dashboards/overview.json:
```diff
@@ -0,0 +1,867 @@
+            "title": true,
+            "title_align": "center",
+            "title_size": 16,
+            "title_text": "#Transactions",
```
```suggestion
            "title_text": "# Transactions",
```
consistent spacing

ChristineTChen commented on nomad/assets/dashboards/overview.json:
```diff
@@ -0,0 +1,867 @@
+            "title": true,
+            "title_align": "center",
+            "title_size": 16,
+            "title_text": "Transaction Time",
+            "type": "query_value",
```
Would transaction time be better as a time series? If there are multiple nomad hosts, a timeseries can show the avg transaction time for a specific period by host.

ChristineTChen commented on nomad/assets/dashboards/overview.json:
```diff
@@ -0,0 +1,867 @@
+        "author_name": "Datadog"
+    },
+    "board_title": "Nomad Overview",
+    "description": "",
```
Can we add a description? See examples in our default dashboards. e.g
![image](https://user-images.githubusercontent.com/15065007/85447875-14e30200-b564-11ea-98db-1e0f32e7344a.png)

steveny91 commented:
Following your feedback, I made some additional adjustments to display graph where it made more sense. Changed the titles of thee query value widgets up a bit. I felt it was more accurate to use the QVW and use the sum aggregation when talking about totals. 

Added a simple description.

New screenshot for the updated dashboard:
https://a.cl.ly/lluJNKod

<img width="582" alt="Image 2020-06-24 at 1 48 13 PM" src="https://user-images.githubusercontent.com/19823071/85607798-7669a600-b622-11ea-9caa-61422b86f413.png">

ChristineTChen commented on nomad/assets/dashboards/overview.json:
```diff
@@ -0,0 +1,978 @@
+            "height": 5,
+            "html": "Runtime Metrics",
+            "text_align": "center",
+            "tick": false,
```
`Runtime Metrics` pointer is not enabled. Can we also shift the widgets down by 1 pixel in this section to show pointer.
```suggestion
            "tick": true,
```

ChristineTChen commented on nomad/assets/dashboards/overview.json:
```diff
@@ -0,0 +1,978 @@
+            "title": true,
+            "title_align": "center",
+            "title_size": 16,
+            "title_text": "Total Brokers Blocked",
```
Can we have 1 pixel between each widget and label for `Broker Metrics` and `RPC Metrics`? 

https://datadoghq.dev/integrations-core/guidelines/dashboards/#widgets

ChristineTChen commented on nomad/assets/dashboards/overview.json:
```diff
@@ -0,0 +1,978 @@
+            "bgcolor": "gray",
+            "font_size": "14",
+            "height": 5,
+            "html": "Raft Metrics",
```
Can we shift widgets under `Raft Metrics` label down by 1 pixel to expose the pointer & keep consistent with spacing?

ChristineTChen commented on nomad/assets/dashboards/overview.json:
```diff
@@ -0,0 +1,978 @@
+                "requests": [
+                    {
+                        "aggregator": "avg",
+                        "q": "avg:nomad.runtime.num_goroutines{$host,$client_id}"
```
Runtime widgets need 1 pixel spacing between graphs.

https://datadoghq.dev/integrations-core/guidelines/dashboards/#widgets
